torch/utils/_pytree/api/cxx.py

```diff
@@ -14,14 +14,8 @@
 
 import functools
 import inspect
-import warnings
 from typing import Any, Callable, Iterable, List, Optional, overload, Tuple, Type, Union
 
-import torch
-
-if torch._running_with_deploy():
-    warnings.warn("C++ Pytree utilities do not work with torch::deploy.")
-
 import optree
 from optree import PyTreeSpec  # direct import for type annotations
```

Comment:
In `torch`, there are many places to import C extensions. They are also guarded with `torch._running_with_deploy()` condition.

https://github.com/pytorch/pytorch/blob/29f3d392bf230072e3bffae37b078e770cae1956/torch/_ops.py#L914-L941

https://github.com/pytorch/pytorch/blob/29f3d392bf230072e3bffae37b078e770cae1956/torch/cuda/__init__.py#L1268-L1288

https://github.com/pytorch/pytorch/blob/29f3d392bf230072e3bffae37b078e770cae1956/torch/distributed/_functional_collectives.py#L20-L33

https://github.com/pytorch/pytorch/blob/29f3d392bf230072e3bffae37b078e770cae1956/torch/distributed/_functional_collectives.py#L613-L630

https://github.com/pytorch/pytorch/blob/29f3d392bf230072e3bffae37b078e770cae1956/torch/distributed/_tensor/__init__.py#L361-L362